cleverhans/loss.py

```diff
@@ -329,3 +334,100 @@ def fprop(self, x, y, **kwargs):
                   "MixUp. LossFeaturePairing may be removed "
                   "on or after 2019-03-06.")
     return loss
+
+
+class SNNLCrossEntropy(CrossEntropy):
```

Comment:
Please add a docstring.
A few things that should go in the docstring:
  - Specification that this stands for Soft Nearest Neighbor + whatever the L stands for
  - Link to the paper

Reply:
The docstring should also include ":param" declarations of the arguments of the `__init__` method